packages/template-retail-react-app/app/contexts/index.js

```diff
@@ -27,23 +29,157 @@ import PropTypes from 'prop-types'
+        })
+        if (findSuccess) {
+            for (let prop in tree) {
+                delete tree[prop]
+            }
+            for (let prop in replacementObj) {
+                tree[prop] = replacementObj[prop]
+            }
+            return tree
+        }
+        const findInChildren = (obj, childrenKey, objToFindBy, replacementObj) => {
+            if (Object.prototype.hasOwnProperty.call(obj, childrenKey)) {
+                for (let i = 0; i < obj[childrenKey].length; i++) {
+                    findKeys.forEach((key) => {
+                        isEqual(obj[childrenKey][i][key], objToFindBy[key])
+                            ? (findSuccess = true)
+                            : (findSuccess = false)
+                    })
+                    if (findSuccess) {
+                        obj[childrenKey][i] = replacementObj
+                        modifiedObj = true
+                        break
+                    }
+                }
+                if (!findSuccess) {
+                    obj[childrenKey].forEach((child) =>
+                        findInChildren(child, childrenKey, objToFindBy, replacementObj)
+                    )
+                }
+            }
+            return obj
+        }
+        findInChildren(tree, childrenKey, objToFindBy, replacementObj)
+        return modifiedObj ? treeToReturn : false
+    }
+
+    const fetchCategoryNode = async (id, depth = 1) => {
+        const STALE_TIME = 10000 // 10 seconds
+        let res
+        // return early if there's an in-flight request or one that is less
+        // than the stale time
+        if (queue[id] === 'loading' || Date.now() < queue[id] + STALE_TIME) {
+            return
+        }
+        setQueue({
+            ...queue,
+            [id]: 'loading'
+        })
+        res = await api.shopperProducts.getCategory({
+            parameters: {
+                id,
+                levels: depth
+            }
+        })
+        const newTree = findAndModifyFirst(
+            root,
+            itemsKey,
+            {id},
+            {
+                ...res,
+                loaded: true,
+                [itemsKey]: res?.[itemsKey]?.map((item) => ({...item, loaded: true}))
+            }
+        )
+        setRoot(newTree)
+        setQueue({
+            ...queue,
+            // store a timestamp for evaluating when to refetch
+            [id]: Date.now()
+        })
+        return res
+    }
+
+    useEffect(() => {
+        // Server side, we only fetch level 0 categories, for performance, here
+        // we request the remaining two levels of category depth
+        root?.[itemsKey]?.forEach(async (cat) => {
+            // check localstorage first for this data to help remediate O(n) server
+            // load burden where n = top level categories
+            const storedCategoryData = JSON.parse(
+                window?.localStorage?.getItem(`${LOCAL_STORAGE_PREFIX + cat?.id}`)
```

Comment:
```suggestion
                window?.localStorage?.getItem(`${LOCAL_STORAGE_PREFIX}${cat?.id}`)
```
```suggestion
                window?.localStorage?.getItem(LOCAL_STORAGE_PREFIX + cat?.id)
```
Pick one or the other, using both is unnecessary. 😉